src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -2150,6 +2157,40 @@ void MacroAssembler::call_VM_leaf_base(address entry_point) {
   call_VM_leaf_base(entry_point, allow_relocation);
 }
 
+int MacroAssembler::ic_check_size() {
+  return 30 + (ImplicitNullChecks ? 0 : 6);
```

Comment:
Did you check if the size is correct and the alignment works as intended in the -XX:+PrintAssembly output with and without -XX:-ImplicitNullChecks?

Reply:
Hi Martin, 

This is output with flag: 
```asm
[Entry Point]
  # {method} {0x000003ff500007d0} 'hashCode' '()I' in 'java/lang/Object'
  #           [sp+0xd0]  (sp of caller)
  0x000003ff8cc30980:   nopr
  0x000003ff8cc30982:   nopr
  0x000003ff8cc30984:   nopr
  0x000003ff8cc30986:   nopr
  0x000003ff8cc30988:   nopr
  0x000003ff8cc3098a:   nopr
  0x000003ff8cc3098c:   nopr
  0x000003ff8cc3098e:   nopr
  0x000003ff8cc30990:   nopr
  0x000003ff8cc30992:   nopr
  0x000003ff8cc30994:   nopr
  0x000003ff8cc30996:   nopr
  0x000003ff8cc30998:   nopr
  0x000003ff8cc3099a:   nopr
  0x000003ff8cc3099c:   cgije   %r2,0,0x000003ff8cc309b2
  0x000003ff8cc309a2:   llgf    %r1,8(%r2)
  0x000003ff8cc309a8:   cg  %r1,8(%r9)
  0x000003ff8cc309ae:   je  0x000003ff8cc309c0
  0x000003ff8cc309b2:   iihf    %r1,1023                    ;const 0x3ff8c51b8c0 | 4396105709760 |   4.396105709760000e+12
  0x000003ff8cc309b8:   iilf    %r1,2354165952
  0x000003ff8cc309be:   br  %r1
```

And this is without ImplicitNullCheck flag: 
```asm
[Entry Point]
  # {method} {0x000003ff540007d0} 'hashCode' '()I' in 'java/lang/Object'
  #           [sp+0xd0]  (sp of caller)
  0x000003ff94c30980:   nopr
  0x000003ff94c30982:   nopr
  0x000003ff94c30984:   nopr
  0x000003ff94c30986:   nopr
  0x000003ff94c30988:   nopr
  0x000003ff94c3098a:   nopr
  0x000003ff94c3098c:   nopr
  0x000003ff94c3098e:   nopr
  0x000003ff94c30990:   nopr
  0x000003ff94c30992:   nopr
  0x000003ff94c30994:   nopr
  0x000003ff94c30996:   nopr
  0x000003ff94c30998:   nopr
  0x000003ff94c3099a:   nopr
  0x000003ff94c3099c:   nopr
  0x000003ff94c3099e:   nopr
  0x000003ff94c309a0:   nopr
  0x000003ff94c309a2:   llgf    %r1,8(%r2)
  0x000003ff94c309a8:   cg  %r1,8(%r9)
  0x000003ff94c309ae:   je  0x000003ff94c309c0
  0x000003ff94c309b2:   iihf    %r1,1023                    ;const 0x3ff9451b8c0 | 4396239927488 |   4.396239927488000e+12
  0x000003ff94c309b8:   iilf    %r1,2488383680
  0x000003ff94c309be:   br  %r1
[Verified Entry Point]
```

Also with latest patch from Erik, `CastNullCheckDroppingsTest.java` test is also passing.